Bumbodium/Models/AvailabilityVM.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Bumbodium.Data;
+using Bumbodium.Data.DBModels;
+using System.ComponentModel.DataAnnotations;
+
+namespace Bumbodium.WebApp.Models
+{
+    public class AvailabilityVM : IValidatableObject
+    {
+        [Required]
+        public string EmployeeName { get; set; }
+
+        [Required]
+        public DateTime StartTime { get; set; }
+
+        [Required]
+        public DateTime EndTime { get; set; }
+
+        [Required]
+        public AvailabilityType Type { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            //Db doohickery
+            var _ctx = (BumbodiumContext)validationContext
+                         .GetService(typeof(BumbodiumContext));
+            var _availabilityRepo = new AvailabilityRepo(_ctx);
+            var _employeeRepo = new EmployeeRepo(_ctx);
+            var _employee = _employeeRepo.GetEmployeeByName(EmployeeName);
+
+            //Verify that availability can only be added if startTime is before EndTime
+            if (EndTime.Subtract(StartTime) < TimeSpan.Zero)
+                yield return new ValidationResult("End time cannot be before start time", new[] { "StartTime" });
+
+            //Verify that user cannot have multiple availabilities on the same time and day
+            var availabilities = _availabilityRepo.GetAvailabilitiesInRange(StartTime, EndTime);
+            if(availabilities.Count > 0)
+                yield return new ValidationResult("Cannot have multiple availabilities at the same time", new[] { "StartTime" });
```

Comment:
GetAvailabilitiesInRange finds all availabilities of which **StartTime** is in the given range. An availability record of which the EndTime is in the range wont be found. You could expand the GetAvailabilitiesInRange method to include records where the EndTime is within the given range.
But then availabilities encapsulating the range wont be found. Thus the issue of redundant availabilities persists.

Reply:
GetAvailabilities gets the availabilities where the **StartTime of the availability** is between StartTime and EndTime

![image](https://user-images.githubusercontent.com/72139731/210828632-c9f5d111-982b-47d1-9fdf-2f8dab1546b1.png)
The availabilities in case 2 and 3 are not found because the StartTime of these availabilities falls outside of the query range

Reply:
`WHERE AStart BETWEEN QStart AND QEnd AND AEnd BETWEEN QStart AND QEnd && (AStart < QStart && AEnd > QStart)`
;)